schema/url/computes/correlation.raml

```diff
@@ -21,11 +21,23 @@ types:
       correlationMethod: CorrelationMethod
       prefilterThresholds?: FeaturePrefilterThresholds
 
+
+  CorrelationInputDataTypes:
+    type: string
+    enum: ["metadata","collection"]
+
+# If metadata isn't a collection, then we shouldn't be forcing correlation to use collections
+# as the input. It should be an option of a data type.
+  CorrelationInputData:
+    properties: 
+      dataType: CorrelationInputDataTypes
+      collectionSpec?: CollectionSpec
+
   CorrelationConfig:
     type: BaseCorrelationComputeConfig
     properties: 
-      collectionVariable: CollectionSpec
-      collectionVariable2?: CollectionSpec
+      data1: CorrelationInputData
```

Comment:
this implies you can pass metadata v metadata here. i dont think thats right?

Reply:
Good question. But why not? I could pass collection A and collection A here. We could still have the selfCorrelation because it's easier to use. But i don't think there's anything incorrect about it. Outside of that i haven't added support for it in the plugin... lol

Reply:
Not just that we haven't actually supported it.. but that I mean self correlation to support different methods (sparcc) and that I also like there being one defined way to do a thing. Idk why we'd want multiple endpoints doing the same thing using different apis.

Reply:
Sounds like we just need the plugin to check to ensure that data1 and data2 are different. No reason why we can't do metadata vs assay, we just want to prevent metadata vs metadata and Assay 1 vs Assay 1.